### PR TITLE
Workspace invite

### PR DESCRIPTION
**Pull request by speed2exe**
- Deprecate direct adding members to workspace.
- In order to add members to workspace, owners will sent an invitation. After receiver accepts invitation, he/she will then be added to the workspace

### CODE REVIEW

File: src/biz/workspace/ops.rs [hunk 1]
```diff
@@ -131,6 +137,60 @@ pub async fn open_workspace(
...
+      &mut txn,
+      workspace_id,
+      inviter,
+      invitation.email.as_str(),
+      invitation.role,
+    )
+    .await?;
+  }
+
+  txn.commit().await?;
```
Comment (appflowy) [inline src/biz/workspace/ops.rs:179]: how about providing the context("") here?

File: libs/database-entity/src/dto.rs [hunk 2]
```diff
@@ -481,4 +481,35 @@ pub struct AFWorkspaceMember {
...
+  pub workspace_id: Uuid,
+  pub workspace_name: Option<String>,
+  pub inviter_email: Option<String>,
+  pub inviter_name: Option<String>,
+  pub status: AFWorkspaceInvitationStatus,
+  pub updated_at: DateTime<Utc>,
+}
+
+#[derive(Serialize, Deserialize, Eq, PartialEq, Debug, Clone)]
+#[repr(i16)]
```
Comment (appflowy) [inline libs/database-entity/src/dto.rs:496]: how about repr(u8) ?

File: tests/access_control/collab_ac_test.rs [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+      )
+      .await?,
+    "stranger should not have write access"
+  );
+
+  Ok(())
+}
+
+#[sqlx::test(migrations = false)]
+async fn test_collab_access_control_send_receive_collab_update(pool: PgPool) -> anyhow::Result<()> {
```
Comment (appflowy) [inline tests/access_control/collab_ac_test.rs:235]: sometimes get pool timeout when using sqlx::test with pool

File: libs/database-entity/src/dto.rs (hunk 2 above)
Comment (speed2exe) [reply libs/database-entity/src/dto.rs:496]: I have tried, smallest integer postgres can support is 16 bits

File: src/biz/workspace/ops.rs (hunk 1 above)
Comment (speed2exe) [reply src/biz/workspace/ops.rs:179]: ok, I'll add it in

**Comment by speed2exe** [comment]
close in favor of https://github.com/AppFlowy-IO/AppFlowy-Cloud/pull/366 instead

**Pull request closed without merging**